pkg/console/tlsconfig/tlsconfig.go

```diff
@@ -154,68 +163,72 @@ func loadCipherSuitesAndMinVersion(configPath string) ([]uint16, uint16, error)
 		return nil, 0, fmt.Errorf("could not load tls config: %w", err)
 	}
 
-	ciphers, minVersion, err := getTlsCiphersAndMinVersion(tlsProfile)
+	ciphers, err := getCipherSuites(tlsProfile.Ciphers)
 	if err != nil {
-		return nil, 0, err
+		return nil, 0, fmt.Errorf("could not get cipher suite numbers: %w", err)
+	}
+
+	minVersion, err := getMinTlsVersion(tlsProfile.MinTLSVersion)
+	if err != nil {
+		return nil, 0, fmt.Errorf("could not get minimum TLS version: %w", err)
 	}
 
 	return ciphers, minVersion, nil
 }
 
-func loadTlsProfile(profilePath string) (*v1alpha1.TlsSecurityProfile, error) {
+func loadTlsProfile(profilePath string) (*v1.TlsProfile, error) {
 	file, err := os.Open(profilePath)
 	if err != nil {
 		return nil, fmt.Errorf("error opening file: %w", err)
 	}
 	// It's ok to ignore error on close, because the file is opened of reading
 	defer func() { _ = file.Close() }()
 
-	result := &v1alpha1.TlsSecurityProfile{}
+	result := &v1.TlsProfile{}
 	err = yaml.NewYAMLToJSONDecoder(file).Decode(result)
 	if err != nil {
 		return nil, fmt.Errorf("error decoding tls config: %w", err)
 	}
 	return result, nil
 }
 
-func getTlsCiphersAndMinVersion(tlsProfile *v1alpha1.TlsSecurityProfile) ([]uint16, uint16, error) {
-	var profile *ocpconfigv1.TLSProfileSpec
-	if tlsProfile.Type == ocpconfigv1.TLSProfileCustomType {
-		if tlsProfile.Custom == nil {
-			return nil, 0, fmt.Errorf("tls profile \"custom\" field is nil")
-		}
-		profile = &tlsProfile.Custom.TLSProfileSpec
-	} else {
-		var exists bool
-		profile, exists = ocpconfigv1.TLSProfiles[tlsProfile.Type]
-		if !exists {
-			return nil, 0, fmt.Errorf("unknown profile type: %s", tlsProfile.Type)
-		}
+func getCipherSuites(cipherNames []string) ([]uint16, error) {
+	if len(cipherNames) == 0 {
+		// nil value has means default cipher suites will be used
+		return nil, nil
 	}
 
-	ciphers := getCipherSuites(profile)
-	minVersion, err := crypto.TLSVersion(string(profile.MinTLSVersion))
-	if err != nil {
-		return nil, 0, err
+	result := make([]uint16, 0, len(cipherNames))
+
+outerLoop:
+	for _, cipherName := range cipherNames {
+		for _, cipherSuite := range tls.CipherSuites() {
+			if cipherName == cipherSuite.Name {
+				result = append(result, cipherSuite.ID)
+				continue outerLoop
```

Comment:
Will a `break` here have the same effect?

Reply:
done. Sometimes I don't see the simple solution :)

Reply:
No, it wouldn't work. I was confused by your comment. 
The `break` will jump out of the inner loop, but still execute the `return` with error. This `continue` jumps to the next iteration of the outer loop skipping the `return`.